test/button_test.js

```diff
@@ -613,6 +615,56 @@ describe("DoorhangerAddToToolbar Treatment Tests", function() {
   });
 });
 
+describe("Expiration date tests", function() {
+  // This gives Firefox time to start, and us a bit longer during some of the tests.
+  this.timeout(15000);
+
+  let driver;
+
+  before(async() => {
+    driver = await utils.promiseSetupDriver();
+    // set expiration date to a date sufficiently in the past to trigger expiration
+    const now = new Date(Date.now());
+    const expiredDateString = new Date(now.setDate(now.getDate() - 30)).toISOString();
+    await driver.executeAsyncScript((dateStringArg, callback) => {
+      Components.utils.import("resource://gre/modules/Preferences.jsm");
+      Preferences.set("extensions.sharebuttonstudy.expirationDateString", dateStringArg);
```

Comment:
`Prefrences.jsm` is sort of deprecated, I think. This should use `Services.prefs.setStringPref` instead.

Reply:
Per our discussion in person, this should be fine.